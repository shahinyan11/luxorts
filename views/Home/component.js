import React from 'react';
import { Form } from 'antd';
import { FormattedMessage } from 'react-intl';
import { FormikProvider } from 'formik';
import Link from 'next/link';

import ROUTES from 'constants/routes';
import { PROPERTY_TYPE_IMAGES, TRENDING_LOCATION_IMAGES } from 'constants';

import RangePickerField from 'views/shared/form/RangePickerField';
import HomePageLayout from 'views/layouts/HomePage';
import GradientButton from 'views/shared/GradientButton';
import InputField from 'views/shared/form/InputField';
import SvgIcon from 'views/shared/SvgIcon';
import TrendingCard from 'views/shared/TrendingCard';
import PropertyCard from 'views/shared/PropertyCard';
import ApartmentCard from 'views/shared/ApartmentCard';

import useContainer, { getInitialProps } from './hook';

const Home = () => {
  const {
    formik,
    propertyTypes,
    trendingLocations,
    popularApartments,
    popularApartmentsLastPage,
    popularApartmentsSelfPage,
    handleNext,
    handlePrev,
  } = useContainer();

  return (
    <>
      <section className="intro">
        <div className="intro__container">
          <h1 className="intro__title mb-16">
            <FormattedMessage id="home.findYour" />
            <span className="intro__title-accent">
              <FormattedMessage id="home.perfectApartments" />
            </span>
          </h1>
          <p className="intro__text mb-40">
            <FormattedMessage id="home.discoverAndBookApartments" />
          </p>
          <Form layout="vertical" size="large" className="intro__search search search--home">
            <FormikProvider value={formik}>
              <div className="search__location">
                <InputField
                  name="location"
                  label={{ id: 'shared.location' }}
                  placeholder={{ id: 'shared.whereAreYouGoing' }}
                  formItemClassName="search__item"
                  className="search__input"
                  allowClear={{ clearIcon: <SvgIcon icon="cross" /> }}
                />
              </div>
              <div className="search__dates">
                <div className="search__labels">
                  <p className="search__label">
                    <FormattedMessage id="shared.checkIn" />
                  </p>
                  <p className="search__label">
                    <FormattedMessage id="shared.checkOut" />
                  </p>
                </div>
                <RangePickerField
                  learIcon={<SvgIcon icon="cross" />}
                  placeholder={[{ id: 'shared.dateIn' }, { id: 'shared.dateOut' }]}
                  suffixIcon={false}
                  separator={false}
                  className="w-100"
                  size="large"
                  name="date"
                />
              </div>
              <div className="search__guests">
                <InputField
                  name="guests"
                  label={{ id: 'shared.guests' }}
                  placeholder={{ id: 'shared.addGuests' }}
                  formItemClassName="search__item mb-0"
                  className="search__input"
                  allowClear={{ clearIcon: <SvgIcon icon="cross" /> }}
                />
                <GradientButton
                  className="search__submit main-btn--icon"
                  variant="primary"
                  addonBefore={<SvgIcon icon="search" className="icon-right" />}
                />
              </div>
            </FormikProvider>
          </Form>
          <img alt="" width="1140" height="380" className="intro__img" src="/images/intro.jpg" />
        </div>
      </section>
      <section className="trending pt-40 pt-md-80 pb-40 pb-md-40">
        <div className="trending__container">
          <h2 className="trending__title mb-8">
            <FormattedMessage id="home.trendingLocationsTitle" />
          </h2>
          <p className="trending__description mb-40">
            <FormattedMessage id="home.trendingLocationsText" />
          </p>
          <div className="trending__pictures">
            {trendingLocations.map((value, index) => (
              <TrendingCard
                key={value.id}
                image={TRENDING_LOCATION_IMAGES[index]}
                small={index > 1}
                {...value}
              />
            ))}
          </div>
        </div>
      </section>
      <section className="property">
        <div className="property__container pt-40 pt-md-80">
          <h2 className="property__title mb-8">
            <FormattedMessage id="home.propertyTypesTitle" />
          </h2>
          <p className="property__description mb-40">
            <FormattedMessage id="home.propertyTypesText" />
          </p>
          <div className="property__cards pb-40 pb-md-80">
            {propertyTypes.map((value, index) => (
              <PropertyCard key={value.id} image={PROPERTY_TYPE_IMAGES[index]} {...value} />
            ))}
          </div>
        </div>
      </section>
      <section className="popular-apartment pt-40 pt-md-80 pb-40 pb-md-80">
        <div className="popular-apartment__container">
          <h2 className="popular-apartment__title mb-8 mb-md-0">
            <FormattedMessage id="home.popularApartmentsTitle" />
          </h2>
          <div className="d-flex flex-column mb-40">
            <p className="popular-apartment__description mb-8 mb-md-0">
              <FormattedMessage id="home.popularApartmentsText" />
            </p>
            <div className="popular-apartment__column d-flex align-items-center ml-md-auto">
              <p className="counter ml-auto mr-16">
                <span className="counter__number counter__number--current">
                  {popularApartmentsSelfPage}
                </span>
                <span className="counter__divider">/</span>
                <span className="counter__number">{popularApartmentsLastPage}</span>
              </p>
              <GradientButton
                onClick={handlePrev}
                className="main-btn--icon main-btn--small mr-16"
                variant="primary"
                addonBefore={<SvgIcon icon="arrow-left" className="icon-right" />}
                disabled={popularApartmentsSelfPage === 1}
              />
              <GradientButton
                onClick={handleNext}
                className="main-btn--icon main-btn--small"
                variant="primary"
                addonBefore={<SvgIcon icon="arrow-right" className="icon-right" />}
                disabled={popularApartmentsSelfPage === popularApartmentsLastPage}
              />
            </div>
          </div>
          <div className="popular-apartment__slider">
            <div className="popular-apartment__slider-wrapper">
              {popularApartments.map((value) => (
                <ApartmentCard key={value.id} {...value} />
              ))}
            </div>
          </div>
        </div>
      </section>
      <section className="cta pt-40 pt-md-80 pb-40 pb-md-80">
        <div className="cta__container">
          <div className="cta__content">
            <h2 className="cta__title mb-8">
              <FormattedMessage
                id="home.tryHosting"
                values={{
                  styledWord: (
                    <span className="cta__accent">
                      <FormattedMessage id="home.hosting" />
                    </span>
                  ),
                }}
              />
            </h2>
            <p className="cta__description mb-32">
              <FormattedMessage id="home.listYourPropertyText" />
            </p>
            <Link href={ROUTES.ADD_NEW_LISTING.INDEX.PATH}>
              <a className="main-btn main-btn--primary min-width-160">
                <FormattedMessage id="home.listYourProperty" />
              </a>
            </Link>
          </div>
        </div>
      </section>
    </>
  );
};

Home.getInitialProps = getInitialProps;

Home.Layout = HomePageLayout;

export default Home;
