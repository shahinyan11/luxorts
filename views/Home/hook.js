import { useFormik } from 'formik';
import { useDispatch, useSelector } from 'react-redux';

import {
  fetchHomePageData,
  fetchPopularApartments,
  setApartmentsPage,
} from 'state/concepts/publicInfo/actions';
import {
  popularApartmentsLastPageSelector,
  popularApartmentsSelfPageSelector,
  propertyTypesSelector,
  trendingLocationsSelector,
} from 'state/concepts/publicInfo/selectors';
import { listingsSelector } from 'state/concepts/listings/selectors';

function useContainer() {
  const dispatch = useDispatch();
  const trendingLocations = useSelector(trendingLocationsSelector);
  const propertyTypes = useSelector(propertyTypesSelector);
  const popularApartments = useSelector(listingsSelector);
  const popularApartmentsLastPage = useSelector(popularApartmentsLastPageSelector);
  const popularApartmentsSelfPage = useSelector(popularApartmentsSelfPageSelector);

  /**
   * Formik initialization
   */
  const formik = useFormik({
    initialValues: {
      location: '',
      date: '',
    },
  });

  /**
   * Handle change to next page
   */
  const handleNext = () => {
    dispatch(setApartmentsPage(popularApartmentsSelfPage + 1));
    dispatch(fetchPopularApartments());
  };

  /**
   * Handle change to next page
   */
  const handlePrev = () => {
    dispatch(setApartmentsPage(popularApartmentsSelfPage - 1));
    dispatch(fetchPopularApartments());
  };

  return {
    formik,
    propertyTypes,
    trendingLocations,
    popularApartments,
    popularApartmentsLastPage,
    popularApartmentsSelfPage,
    handleNext,
    handlePrev,
  };
}

export const getInitialProps = async (ctx) => {
  ctx.store.dispatch(fetchHomePageData());
};

export default useContainer;
