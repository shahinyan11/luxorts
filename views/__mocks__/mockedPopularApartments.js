export default [
  {
    id: '06a30f07-23b5-4900-b83c-b56986f119bc',
    bedroomsAmount: 1,
    bathsAmount: 28,
    propertyTypeMame: 'voluptatum',
    title: 'Alta Kling',
    guestsNumber: 3,
    bedsAmount: 10,
    description: 'Vanessa Swift',
    typeOfPlace: 'shared_room',
    pricePerDay: 6860000,
    limitedListingPhotos: [
      {
        id: '115e2205-856f-4ed4-9f17-1dd6f37f7520',
        photoUrls: {
          small:
            'https://luxorts-backend-staging.s3.us-east-2.amazonaws.com/uploads/listingphoto/115e2205-856f-4ed4-9f17-1dd6f37f7520/photo/small-cf3518983b1ecf97422a2bda10eca45b.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAXWPQSY4555KWNWFP%2F20220717%2Fus-east-2%2Fs3%2Faws4_request&X-Amz-Date=20220717T094425Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=128ecc291ddb112fec4d57ff3e1fb7b91249045e54e653d8b083113e9b9115a7',
        },
      },
    ],
  },
];
