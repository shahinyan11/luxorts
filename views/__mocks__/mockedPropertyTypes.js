export default [
  {
    id: 'd46e3bec-bb9a-4574-ba6d-1a3e19332b5a',
    name: 'totam',
    listingsCount: 1,
  },
];
