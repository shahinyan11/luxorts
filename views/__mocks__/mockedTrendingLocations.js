export default [
  {
    id: 'united-states-pennsylvania-mineral-point',
    country: 'United States',
    city: 'Baltimore',
    state: 'Maryland',
  },
];
