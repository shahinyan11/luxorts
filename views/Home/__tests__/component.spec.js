import { shallow } from 'enzyme';

import mockedPopularApartments from 'views/__mocks__/mockedPopularApartments';
import mockedTrendingLocations from 'views/__mocks__/mockedTrendingLocations';
import mockedPropertyTypes from 'views/__mocks__/mockedPropertyTypes';

import Home from '../component';

jest.mock('state/concepts/publicInfo/selectors', () => ({
  propertyTypesSelector: jest.fn(() => mockedPropertyTypes),
  trendingLocationsSelector: jest.fn(() => mockedTrendingLocations),
  popularApartmentsLastPageSelector: jest.fn(() => 7),
  popularApartmentsSelfPageSelector: jest.fn(() => 1),
}));

jest.mock('state/concepts/listings/selectors', () => ({
  listingsSelector: jest.fn(() => mockedPopularApartments),
}));

describe('Home component', () => {
  const component = shallow(<Home />);

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
