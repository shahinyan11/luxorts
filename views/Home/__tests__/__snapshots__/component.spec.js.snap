// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<Fragment>
  <section
    className="intro"
  >
    <div
      className="intro__container"
    >
      <h1
        className="intro__title mb-16"
      >
        <Memo(FormattedMessage)
          id="home.findYour"
        />
        <span
          className="intro__title-accent"
        >
          <Memo(FormattedMessage)
            id="home.perfectApartments"
          />
        </span>
      </h1>
      <p
        className="intro__text mb-40"
      >
        <Memo(FormattedMessage)
          id="home.discoverAndBookApartments"
        />
      </p>
      <ForwardRef(InternalForm)
        className="intro__search search search--home"
        layout="vertical"
        size="large"
      >
        <ContextProvider
          value={
            Object {
              "dirty": true,
              "handleSubmit": [MockFunction],
              "initialValues": Object {
                "date": "",
                "location": "",
              },
              "isValid": true,
              "resetForm": [MockFunction],
              "setFieldTouched": [MockFunction],
              "setFieldValue": [MockFunction],
              "setValues": [MockFunction],
              "validateForm": [MockFunction],
              "values": Object {},
            }
          }
        >
          <div
            className="search__location"
          >
            <InputField
              allowClear={
                Object {
                  "clearIcon": <SvgIcon
                    icon="cross"
                  />,
                }
              }
              asComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "Group": [Function],
                  "Password": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "action": "click",
                      "iconRender": [Function],
                      "visibilityToggle": true,
                    },
                    "render": [Function],
                  },
                  "Search": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "TextArea": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "render": [Function],
                }
              }
              className="search__input"
              formItemClassName="search__item"
              label={
                Object {
                  "id": "shared.location",
                }
              }
              name="location"
              numbersOnly={false}
              placeholder={
                Object {
                  "id": "shared.whereAreYouGoing",
                }
              }
              withFormItem={true}
            />
          </div>
          <div
            className="search__dates"
          >
            <div
              className="search__labels"
            >
              <p
                className="search__label"
              >
                <Memo(FormattedMessage)
                  id="shared.checkIn"
                />
              </p>
              <p
                className="search__label"
              >
                <Memo(FormattedMessage)
                  id="shared.checkOut"
                />
              </p>
            </div>
            <RangePickerField
              className="w-100"
              learIcon={
                <SvgIcon
                  icon="cross"
                />
              }
              name="date"
              placeholder={
                Array [
                  Object {
                    "id": "shared.dateIn",
                  },
                  Object {
                    "id": "shared.dateOut",
                  },
                ]
              }
              separator={false}
              size="large"
              suffixIcon={false}
            />
          </div>
          <div
            className="search__guests"
          >
            <InputField
              allowClear={
                Object {
                  "clearIcon": <SvgIcon
                    icon="cross"
                  />,
                }
              }
              asComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "Group": [Function],
                  "Password": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "action": "click",
                      "iconRender": [Function],
                      "visibilityToggle": true,
                    },
                    "render": [Function],
                  },
                  "Search": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "TextArea": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "render": [Function],
                }
              }
              className="search__input"
              formItemClassName="search__item mb-0"
              label={
                Object {
                  "id": "shared.guests",
                }
              }
              name="guests"
              numbersOnly={false}
              placeholder={
                Object {
                  "id": "shared.addGuests",
                }
              }
              withFormItem={true}
            />
            <GradientButton
              addonAfter={null}
              addonBefore={
                <SvgIcon
                  className="icon-right"
                  icon="search"
                />
              }
              className="search__submit main-btn--icon"
              variant="primary"
            />
          </div>
        </ContextProvider>
      </ForwardRef(InternalForm)>
      <img
        alt=""
        className="intro__img"
        height="380"
        src="/images/intro.jpg"
        width="1140"
      />
    </div>
  </section>
  <section
    className="trending pt-40 pt-md-80 pb-40 pb-md-40"
  >
    <div
      className="trending__container"
    >
      <h2
        className="trending__title mb-8"
      >
        <Memo(FormattedMessage)
          id="home.trendingLocationsTitle"
        />
      </h2>
      <p
        className="trending__description mb-40"
      >
        <Memo(FormattedMessage)
          id="home.trendingLocationsText"
        />
      </p>
      <div
        className="trending__pictures"
      >
        <TrendingCard
          city="Baltimore"
          country="United States"
          id="united-states-pennsylvania-mineral-point"
          image="/images/trending-location-01.jpg"
          key="united-states-pennsylvania-mineral-point"
          small={false}
          state="Maryland"
        />
      </div>
    </div>
  </section>
  <section
    className="property"
  >
    <div
      className="property__container pt-40 pt-md-80"
    >
      <h2
        className="property__title mb-8"
      >
        <Memo(FormattedMessage)
          id="home.propertyTypesTitle"
        />
      </h2>
      <p
        className="property__description mb-40"
      >
        <Memo(FormattedMessage)
          id="home.propertyTypesText"
        />
      </p>
      <div
        className="property__cards pb-40 pb-md-80"
      >
        <PropertyCard
          id="d46e3bec-bb9a-4574-ba6d-1a3e19332b5a"
          image="/images/property-01.jpg"
          key="d46e3bec-bb9a-4574-ba6d-1a3e19332b5a"
          listingsCount={1}
          name="totam"
        />
      </div>
    </div>
  </section>
  <section
    className="popular-apartment pt-40 pt-md-80 pb-40 pb-md-80"
  >
    <div
      className="popular-apartment__container"
    >
      <h2
        className="popular-apartment__title mb-8 mb-md-0"
      >
        <Memo(FormattedMessage)
          id="home.popularApartmentsTitle"
        />
      </h2>
      <div
        className="d-flex flex-column mb-40"
      >
        <p
          className="popular-apartment__description mb-8 mb-md-0"
        >
          <Memo(FormattedMessage)
            id="home.popularApartmentsText"
          />
        </p>
        <div
          className="popular-apartment__column d-flex align-items-center ml-md-auto"
        >
          <p
            className="counter ml-auto mr-16"
          >
            <span
              className="counter__number counter__number--current"
            >
              1
            </span>
            <span
              className="counter__divider"
            >
              /
            </span>
            <span
              className="counter__number"
            >
              7
            </span>
          </p>
          <GradientButton
            addonAfter={null}
            addonBefore={
              <SvgIcon
                className="icon-right"
                icon="arrow-left"
              />
            }
            className="main-btn--icon main-btn--small mr-16"
            disabled={true}
            onClick={[Function]}
            variant="primary"
          />
          <GradientButton
            addonAfter={null}
            addonBefore={
              <SvgIcon
                className="icon-right"
                icon="arrow-right"
              />
            }
            className="main-btn--icon main-btn--small"
            disabled={false}
            onClick={[Function]}
            variant="primary"
          />
        </div>
      </div>
      <div
        className="popular-apartment__slider"
      >
        <div
          className="popular-apartment__slider-wrapper"
        >
          <ApartmentCard
            bathsAmount={28}
            bedroomsAmount={1}
            bedsAmount={10}
            description="Vanessa Swift"
            guestsNumber={3}
            id="06a30f07-23b5-4900-b83c-b56986f119bc"
            key="06a30f07-23b5-4900-b83c-b56986f119bc"
            limitedListingPhotos={
              Array [
                Object {
                  "id": "115e2205-856f-4ed4-9f17-1dd6f37f7520",
                  "photoUrls": Object {
                    "small": "https://luxorts-backend-staging.s3.us-east-2.amazonaws.com/uploads/listingphoto/115e2205-856f-4ed4-9f17-1dd6f37f7520/photo/small-cf3518983b1ecf97422a2bda10eca45b.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAXWPQSY4555KWNWFP%2F20220717%2Fus-east-2%2Fs3%2Faws4_request&X-Amz-Date=20220717T094425Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=128ecc291ddb112fec4d57ff3e1fb7b91249045e54e653d8b083113e9b9115a7",
                  },
                },
              ]
            }
            pricePerDay={6860000}
            propertyTypeMame="voluptatum"
            title="Alta Kling"
            typeOfPlace="shared_room"
          />
        </div>
      </div>
    </div>
  </section>
  <section
    className="cta pt-40 pt-md-80 pb-40 pb-md-80"
  >
    <div
      className="cta__container"
    >
      <div
        className="cta__content"
      >
        <h2
          className="cta__title mb-8"
        >
          <Memo(FormattedMessage)
            id="home.tryHosting"
            values={
              Object {
                "styledWord": <span
                  className="cta__accent"
                >
                  <Memo(FormattedMessage)
                    id="home.hosting"
                  />
                </span>,
              }
            }
          />
        </h2>
        <p
          className="cta__description mb-32"
        >
          <Memo(FormattedMessage)
            id="home.listYourPropertyText"
          />
        </p>
        <Link
          href="/add-new-listing"
        >
          <a
            className="main-btn main-btn--primary min-width-160"
          >
            <Memo(FormattedMessage)
              id="home.listYourProperty"
            />
          </a>
        </Link>
      </div>
    </div>
  </section>
</Fragment>
`;
